libcodechecker/libclient/credential_manager.py

```diff
@@ -71,15 +80,42 @@ def get_token(self, host, port):
         return self.__tokens.get("{0}:{1}".format(host, port))
 
     def get_auth_string(self, host, port):
-        ret = self.__save['credentials'].get('{0}:{1}'.format(host, port))
-        if not ret:
-            ret = self.__save['credentials'].get(host)
-        if not ret:
-            ret = self.__save['credentials'].get('*:{0}'.format(port))
-        if not ret:
-            ret = self.__save['credentials'].get('*')
-
-        return ret
+        """
+        hostname without protocol and product url
+        """
+        p = re.compile(
+            r'^(?P<protocol>http[s]?://)*(?P<host>[\w.\*]+)'
+            r'/*(?P<product>\w+)*:*(?P<port>\d+)*$')
+
+        full_match = None
+        host_only_match = None
+        all_host_port = None
+        all_host_no_port = None
+
+        for host_entry, auth_string in self.__save['credentials'].items():
+            match = p.match(host_entry)
+            if match:
+                host_match = match.group('host')
+                port_match = match.group('port')
+                if host_match == host and port == port_match:
+                    full_match = auth_string
```

Comment:
In case of full match we can return immediately.

Reply:
I would keep the return logic to the end at one place so the return conditions are visible there together.